### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by cjstehno**
We had some questions about logging recently, so I added some logging to the core client and some documentation around logging in general.

@dwclark please give this a once-over when you have time - mainly interested in the logging I added to `JavaHttpBuilder`. Since these are at DEBUG levels they should not pose much of a performance hit, but I wanted to be sure this would not effect your use cases.

### CODE REVIEW

File: http-builder-ng-core/src/main/java/groovyx/net/http/NativeHandlers.java [hunk 1]
```diff
@@ -348,28 +354,6 @@ public static void json(final ChainedHttpConfig config, final ToServer ts) {
             }
         }
 
-        public static void transfer(final InputStream istream, final OutputStream ostream, final boolean close) {
```
Comment (dwclark) [inline http-builder-ng-core/src/main/java/groovyx/net/http/NativeHandlers.java]: Can we resurrect this method, forward calls on to IoUtils.transfer(), and then deprecate it? I doubt anyone was using it, but it is public so it's possible someone was using it. We can let it live for another minor version or two and then kill it. We can definitely kill it for a 1.0 release.

File: http-builder-ng-core/src/test/resources/log4j.xml [hunk 2]
```diff
@@ -1,36 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dwclark) [inline http-builder-ng-core/src/test/resources/log4j.xml]: Hooray! Death to XML!

File: http-builder-ng-core/src/main/java/groovyx/net/http/JavaHttpBuilder.java [hunk 3]
```diff
@@ -92,58 +109,84 @@ private PasswordAuthentication getAuthInfo() {
...
             void transfer() throws IOException {
-                NativeHandlers.Parsers.transfer(inputStream, connection.getOutputStream(), true);
+                IoUtils.transfer(inputStream, connection.getOutputStream(), true);
+            }
+
+            public String content() {
+                try {
+                    return IoUtils.copyAsString(inputStream);
+                } catch (IOException ioe) {
+                    log.warn("Unable to render request stream due to error (may not affect actual content)", ioe);
```
Comment (dwclark) [inline http-builder-ng-core/src/main/java/groovyx/net/http/JavaHttpBuilder.java]: Nitpick, but let's guard all logging statements for consistency. I know I will add some string interpolation crap here and then forget to add the guards.

File: http-builder-ng-core/src/main/java/groovyx/net/http/JavaHttpBuilder.java [hunk 4]
```diff
@@ -159,38 +202,47 @@ public JavaFromServer(final URI originalUri) throws IOException {
                 is = (bis == null) ? null : handleEncoding(bis);
             }
 
+            String content() {
+                try {
+                    return IoUtils.copyAsString(is);
+                } catch (IOException ioe) {
+                    log.warn("Unable to render response stream due to error (may not affect actual content)", ioe);
```
Comment (dwclark) [inline http-builder-ng-core/src/main/java/groovyx/net/http/JavaHttpBuilder.java]: Same as previous comment for logging guards.

File: http-builder-ng-core/src/main/java/groovyx/net/http/NativeHandlers.java (hunk 1 above)
Comment (cjstehno) [inline http-builder-ng-core/src/main/java/groovyx/net/http/NativeHandlers.java]: Ah, yeah, good call... I got a bit too aggressive. :-)

**Pull request merged**